pkgs/development/haskell-modules/configuration-darwin.nix

```diff
@@ -175,6 +175,19 @@ self: super: ({
       configureFlags+=$(if [ -n "$frameworkPaths" ]; then echo -n "--ghc-options=-optl=$frameworkPaths"; fi)
     '' + (drv.preConfigure or "");
   }) super.OpenGLRaw;
+
+  posix-socket = overrideCabal (drv: {
+    # By default, the wrong uio.h is picked up which doesn't expose the UIO_MAXIOV constant.
+    # This links the correct internal file that contains this constant
+    # The upstream version contains code that doesn't compile but is hidden behind compiler
+    # directives that are always set on Linux but not on darwin. Removing the lines containing
+    # the invalid code actually makes it compile and work
+    postPatch = ''
+      substituteInPlace src/System/Posix/Socket.hsc --replace-fail '#include <sys/uio.h>' '#include "${pkgs.apple-sdk}/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/System/Library/Frameworks/Kernel.framework/Headers/sys/uio.h"'
```

Comment:
seems like just the sdk needs to be upgraded? This way of fixing it seems problematic since the sdk versions aren't the same on all darwin platforms so the path wouldn't necessarily exist.

cc @NixOS/darwin-maintainers

Reply:
Yeah this isn’t right for `x86_64-darwin` on 24.11 and it’s not really what you should be doing in general. `"$include <$SDKROOT/System/Library/Frameworks/Kernel.framework/Header/sys/uio.h>"` would be the “correct” hack, but this looks more like we’re holding `hsc2c` wrong. Is there an easy way of seeing what flags we’re ultimately passing to the C compiler?

Reply:
I guess via configureFlags and `--ghc-option=-v` or so. You can do a lot with e.g. `--extra-include-directories` via Cabal, I think. Depends on what turns out to be the problem.